src/ui/MentionToolbar.js

```diff
@@ -1,28 +1,46 @@
 export var MentionToolbar;
 
 MentionToolbar = function (refs) {
+    var appManager = refs.appManager;
+    var i18n = refs.i18nManager.get();
 
-    var appManager = refs.appManager,
-    i18nManager = refs.i18nManager,
-    uiManager = refs.uiManager;
-    
-    var i18n = i18nManager.get();
+    var canUserAccessCurrentFavorite = function(user) {
```

Comment:
Is this check being repeated for the same user & favorite? If so, then it would be a good opportunity to memoize this function. Shouldn't take too long to do that.

Reply:
As we discussed, it could be done, only that we would be using old data, not the new permissions data from the request.